llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -4821,9 +4822,16 @@ getAppleRuntimeUnrollPreferences(Loop *L, ScalarEvolution &SE,
         const SCEV *PtrSCEV = SE.getSCEV(Ptr);
         if (SE.isLoopInvariant(PtrSCEV, L))
           continue;
-        if (isa<LoadInst>(&I))
-          LoadedValues.insert(&I);
-        else
+        if (isa<LoadInst>(&I)) {
+          LoadedValuesPlus.insert(&I);
+          // Included 1st users of loaded values
+          for (auto *U : I.users()) {
+            auto *Inst = cast<Instruction>(U);
+            if (!Inst)
+              continue;
```

Comment:
```suggestion
```

This isn't needed with cast, `cast` will assert that the argument can be casted with assertions.